### PR TITLE
[prim] move batch_norm prim test to op_test

### PR DESCRIPTION
**Pull request by cyber-pioneer**
### PR types
Others
### PR changes
Others

### Description
Pcard-66975
Move batch_norm prim-relative test caces to op_test.
And support bfloat16 in composite batch_norm.

### CODE REVIEW

File: test/legacy_test/test_batch_norm_op_v3.py [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+        running_variance,
+        weight,
+        bias,
+        training=not is_test,
+        momentum=momentum,
+        epsilon=epsilon,
+        data_format=data_format,
+        use_global_stats=use_global_stats,
+    )
+    z = F.relu(y)
```
Comment (Charles-hit) [inline test/legacy_test/test_batch_norm_op_v3.py]: why add this?

File: test/legacy_test/test_batch_norm_op_v3.py [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+        self.python_out_sig = ["Y"]
+        self.initConfig()
+        self.initTestCase()
+
+    def test_check_output(self):
+        if self.dtype not in ("uint16", "float16"):
+            self.check_output_with_place(
+                core.CPUPlace(),
+                no_check_set=None,
+                atol=self.fw_comp_atol,
```
Comment (Charles-hit) [inline test/legacy_test/test_batch_norm_op_v3.py]: atol and rtol should be set in setUp function for prim check.

File: test/legacy_test/test_batch_norm_op_v3.py [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+                rtol=self.fw_comp_rtol,
+                check_prim=True,
+                only_check_prim=True,
+            )
+        if paddle.is_compiled_with_cuda():
+            self.check_output_with_place(
+                core.CUDAPlace(0),
+                no_check_set=None,
+                atol=self.fw_comp_atol,
```
Comment (Charles-hit) [inline test/legacy_test/test_batch_norm_op_v3.py]: the same as above

File: test/legacy_test/test_batch_norm_op_v3.py [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+                ['Y'],
+                user_defined_grad_outputs=self.out_grad,
+                check_prim=True,
+                only_check_prim=True,
+            )
+
+    def test_check_grad_scale_bias(self):
+        self.enable_cinn = False
+        self.rev_comp_atol = 1e-3
+        self.rev_comp_rtol = 1e-3
```
Comment (Charles-hit) [inline test/legacy_test/test_batch_norm_op_v3.py]: why?

**Review by Charles-hit — COMMENTED** [review]
change test file name for prim.

File: test/legacy_test/test_batch_norm_op_v3.py (hunk 1 above)
Comment (cyber-pioneer) [reply test/legacy_test/test_batch_norm_op_v3.py]: Change the data distribution of output to make x_grad not too small.

**Comment by cyber-pioneer** [comment]
> change test file name for prim.

done

File: test/legacy_test/test_batch_norm_op_v3.py (hunk 2 above)
Comment (cyber-pioneer) [reply test/legacy_test/test_batch_norm_op_v3.py]: done

File: test/legacy_test/test_batch_norm_op_v3.py (hunk 3 above)
Comment (cyber-pioneer) [reply test/legacy_test/test_batch_norm_op_v3.py]: done

File: test/legacy_test/test_batch_norm_op_v3.py (hunk 4 above)
Comment (cyber-pioneer) [reply test/legacy_test/test_batch_norm_op_v3.py]: origin kernel's grad of scale, bias differ with numpy under 1e-3

File: test/legacy_test/CMakeLists.txt [hunk 5]
```diff
@@ -960,6 +961,7 @@ else()
   set_tests_properties(test_conv3d_transpose_op PROPERTIES TIMEOUT 120)
   set_tests_properties(test_conv3d_op PROPERTIES TIMEOUT 120)
   set_tests_properties(test_norm_op PROPERTIES TIMEOUT 120)
+  set_tests_properties(test_batch_norm_op_prim PROPERTIES TIMEOUT 250)
```
Comment (xiaoguoguo626807) [inline test/legacy_test/CMakeLists.txt:964]: 两个好像一样？

File: test/legacy_test/test_batch_norm_op_prim.py [hunk 6]
```diff
@@ -0,0 +1,521 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (xiaoguoguo626807) [inline test/legacy_test/test_batch_norm_op_prim.py:1]: 这个单测名字可以不加prim,对齐其他op 单测吧。 2018-2023

**Review by xiaoguoguo626807 — APPROVED** [review]
LGTM for composite backward api

File: test/legacy_test/test_batch_norm_op_prim.py (hunk 6 above)
Comment (cyber-pioneer) [reply test/legacy_test/test_batch_norm_op_prim.py:1]: 这个单侧比较特殊，只看护了组合算子，为了区分，添加了prim后缀

File: test/legacy_test/CMakeLists.txt (hunk 5 above)
Comment (cyber-pioneer) [reply test/legacy_test/CMakeLists.txt:964]: 对应不同的平台，设置不同的时间阈值

**Review by Charles-hit — APPROVED** [review]

**Review by ZzSean — APPROVED** [review]
LGTM for `skipIf`

**Review by XieYunshen — APPROVED** [review]
LGTM
单测超时时间设置

**Pull request merged**